executor/src/witgen/affine_expression.rs

```diff
@@ -426,14 +407,10 @@ where
     T: FieldElement,
 {
     fn eq(&self, other: &Self) -> bool {
-        if self.offset != other.offset {
+        if self.offset() != other.offset() {
             return false;
         };
-        let mut self_coeff = self.nonzero_coefficients().to_vec();
-        self_coeff.sort_unstable();
-        let mut other_coeff = other.nonzero_coefficients().to_vec();
-        other_coeff.sort_unstable();
-        self_coeff == other_coeff
+        self.nonzero_coefficients().eq(other.nonzero_coefficients())
```

Comment:
The main problem here is that the One and the Many cases overlap, right? Does it make sense to force Many to have at least two variables?

Reply:
`nonzero_coefficients()` unifies both cases. I did do the conversion down to One, but it was a lil bit slower and the code was not simpler

Reply:
the thing is that `One` avoids allocating a vec![] for a single variable case (basically every constructed expr starts as a single variable), but if we already have the vec![] it's not worth it to get rid of it